lib/atlassian-jwt-authentication/filters.rb

```diff
@@ -134,9 +134,9 @@ def _verify_jwt(addon_key, consider_param: false, skip_qsh_verification: false,
       jwt_verification.exclude_qsh_params = exclude_qsh_params
       jwt_verification.logger = logger if defined?(logger)
 
-      jwt_auth, account_id, context, qsh_verified = jwt_verification.verify
+      jwt_auth, account_id, context, qsh_verified, verified = jwt_verification.verify
```

Comment:
Too many *verified variables 🙈 

Can we differentiate them somehow with more meaningful names?

Reply:
Hard to say for me. Maybe another developer should make his/her hands dirty here and propose improvements.

The `jwt_verification.verify` - that guy is returning something representing verification status/result. I.e.: 
```
verification_result = jwt_verification.verify

if verification_result.failed(skip_qsh_validation)
  render_unauthorized
  return false
end

self.current_jwt_token = verification_result.jwt_auth
self.current_account_id = verification_result.account_id
self.current_jwt_context = verification_result.context
```
...but that would require creating a new class for that `verification_result`. And adjusting tests. And performing manual tests if Clockwork can still install under various scenarios. I don't know if I/we want to go that route...

Reply:
I'm not thinking about introducing some result object, but just changing the name. Looking at the code I don't get what the second verified is for.

Reply:
It's the overall verification result (true/false). So...

```
jwt_auth, account_id, context, qsh_verified, verification_result = jwt_verification.verify
```
Would that make more sense to you?

Reply:
Doesn't make a lot of difference. I'm leaving that to you. My brain hurts when looking at it. But I get that you want to get rid of this hot potato.

Reply:
Yeah, exactly. My brains also hurts whenever I work with a-j-a library. It sure would benefit from refactoring or two. 

One of the reasons we don't do that is that a-j-a is not covered by tests enough. The only test in this repo is:
```
describe "RSpec" do
  it "actually works" do
    expect(1).to eq(1)
  end
end
```
:clap: :clap: :clap: :clapper:

And we have some coverage added with our tests in CLK repo. But they don't cover every possible combination and edge cases (think of double VerifyJwtToken middlewares used in the same app, think of various flows of installation/uninstallation/installation combined with free/pro versions). They don't cover even majority of flows I think.
 
So, to make refactoring a-j-a convenient and safe, we would have to put tremendous effort first to prepare a-j-a for such at-hoc refactorings. I mean, add a full blown suite of tests that would cover our asses in 99% cases. I don't want to lead that effort - it seems to me that more valuable to our customers will be if we just fix/plumb whatever we need in a-j-a and get back to our normal work ASAP.